g11n-ws/vip-manager-l10n/src/main/java/com/vmware/l10n/source/service/impl/SyncI18nSourceServiceImpl.java

```diff
@@ -0,0 +1,171 @@
+/*
+ * Copyright 2019-2022 VMware, Inc.
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package com.vmware.l10n.source.service.impl;
+
+import java.io.File;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Map.Entry;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Service;
+import org.springframework.util.StringUtils;
+
+import com.vmware.l10n.source.service.SyncI18nSourceService;
+import com.vmware.l10n.utils.DiskQueueUtils;
+import com.vmware.vip.api.rest.APIParamName;
+import com.vmware.vip.api.rest.APIV2;
+import com.vmware.vip.common.constants.ConstantsKeys;
+import com.vmware.vip.common.exceptions.VIPHttpException;
+import com.vmware.vip.common.http.HTTPRequester;
+import com.vmware.vip.common.l10n.exception.L10nAPIException;
+import com.vmware.vip.common.l10n.source.dto.ComponentSourceDTO;
+
+/**
+ * 
+ * This class used to send the collection source to singleton server
+ *
+ */
+
+@Service
+public class SyncI18nSourceServiceImpl implements SyncI18nSourceService {
+	
+	private final static Logger LOGGER = LoggerFactory.getLogger(SyncI18nSourceServiceImpl.class);
+	private final static String LOCAL_STR = "local";
+	private final static String I18N_STR = "i18n";
+
+	/** the path of local resource file,can be configured in spring config file **/
+	@Value("${source.bundle.file.basepath}")
+	private String basePath;
+
+	/** the url of Singleton API,can be configed in spring config file **/
+	@Value("${vip.server.url}")
+	private String remoteVIPURL;
+
+	@Value("${vip.server.authentication.enable}")
+	private boolean remoteVIPAuthEnable;
+
+	@Value("${vip.server.authentication.appId:#}")
+	private String remoteVIPAuthAppId;
+
+	@Value("${vip.server.authentication.token:#}")
+	private String remoteVIPAuthAppToken;
+
+	/**
+	 * it represents the remote singleton service is available.
+	 */
+	private boolean singletonConnected = false;
+
+	/**
+	 * 
+	 * Synchronize the updated source to remote Singleton server
+	 *
+	 */
+	@Override
+	public synchronized void sendSourceToI18n() {
+		if (LOCAL_STR.equalsIgnoreCase(remoteVIPURL)) {
+			return;
+		}
+		try {
+			pingSingleton(this.remoteVIPURL);
+			setSingletonConnected(true);
+			processSingletonQueueFiles();
+		} catch (L10nAPIException e) {
+
```

Comment:
put 'setSingletonConnected(false);' in catch section so that it won't trigger sending if not connected.

Reply:
done